src/context/SuperAdminContext.tsx

```diff
@@ -0,0 +1,136 @@
+  setUserIDToConnectAnotherAdmin: (value: User["id"]) => void;
+  makeChangesForSelectedUser: UserObjectForSuperAdmin | undefined;
+  updateUserAfterChanges: (value: User) => void;
+};
+
+const updateUserArray = (arrayToUpdate: Array<User>, changedUser: User) => {
+  const index = findIndex(arrayToUpdate, ["id", changedUser.id]);
+  if (index !== -1) {
+    const newArray = [...arrayToUpdate];
+    newArray.splice(index, 1, changedUser);
+    return newArray;
+  } else arrayToUpdate;
+};
+
+const SuperAdminContext = React.createContext<
+  SuperAdminContextType | undefined
+>(undefined);
+
+export const useSuperAdminFunction = () => {
+  return useContext(SuperAdminContext);
+};
+
+export const SuperAdminProvider = ({ children }) => {
+  const [allUsersInSystem, setAllUsersInSystem] = useState<User[] | undefined>(
+    [],
+  );
+  const [getAllUsers, setGetAllUsers] = useState(false);
+  const [userLevel, setUserLevel] = useState<Role | undefined>(undefined);
+  const [allAdminsAnsSuperAdmins, setAllAdminsAnsSuperAdmins] = useState<
+    User[] | undefined
+  >(undefined);
+  const [makeChangesForSelectedUser, setMakeChangesForSelectedUser] = useState<
+    UserObjectForSuperAdmin | undefined
+  >(undefined);
+  const [buttonToSaveChanhgesPressed, setButtonToSaveChanhgesPressed] =
+    useState(false);
+  const [userIDToConnectAnotherAdmin, setUserIDToConnectAnotherAdmin] =
+    useState<User["id"]>("");
+
+  useEffect(() => {
+    if (getAllUsers === true && userLevel === Role.superadmin) {
+      const getAllUsersThatExistInTheSystem = async () => {
+        try {
+          let allUsers = await getAllUsersData();
+          setAllUsersInSystem(allUsers);
+        } catch (error) {
+          console.log("SuperAdminContext errorMessage ", error);
+        }
+      };
+
+      getAllUsersThatExistInTheSystem();
+    }
+  }, [getAllUsers]);
+
+  const updateUserAfterChanges = (changedUser: User) => {
+    if (makeChangesForSelectedUser?.user.id === changedUser.id) {
+      setMakeChangesForSelectedUser(
+        set(makeChangesForSelectedUser, "user", changedUser),
+      );
+    }
+
+    makeChangesForSelectedUser?.arrayOfUsersIfAdmin &&
+      setMakeChangesForSelectedUser(
+        set(
+          makeChangesForSelectedUser,
+          "arrayOfUsersIfAdmin",
+          updateUserArray(
+            makeChangesForSelectedUser.arrayOfUsersIfAdmin,
+            changedUser,
+          ),
+        ),
+      );
+
+    allUsersInSystem &&
+      setAllUsersInSystem(updateUserArray(allUsersInSystem, changedUser));
+  };
+
+  useEffect(() => {
+    allUsersInSystem && findAdminsAndSuperAdmins(allUsersInSystem);
+  }, [allUsersInSystem]);
+
+  const findAdminsAndSuperAdmins = (userArray: Array<User>) => {
+    let adminArray: Array<User> = [];
+    for (let i = 0; i < userArray.length; i++) {
+      if (userArray[i].role === "admin" || userArray[i].role === "superadmin") {
+        adminArray.push(userArray[i]);
+      }
+    }
```

Comment:
Instead of a for-loop I'd use the `Array.filter()` function.

Example
```tsx
const adminArray = userArray.filter(user => user.role === "admin" || user.role === "superadmin");
```